### PR TITLE
Add luacheck.sh to automate checking all DFHack scripts.

### PR DESCRIPTION
**Pull request by BenLubar**

### CODE REVIEW

**Comment by pronvit** [comment]
Hey, does it produce any useful result for DFHack scripts? I only developed it for Remote scripts and added features I needed there. Also, did you find all types of annotations it supports?

**Comment by BenLubar** [comment]
I'm working on it for this PR: https://github.com/DFHack/dfhack/pull/1295

It's already found a few problems with typo'd variable and function names.

File: index.js [hunk 1]
```diff
@@ -526,133 +789,509 @@ function fntype(call, ctx) {
...
+
+	if (fnname == 'df.reinterpret_cast') {
+		return argTypes[0] == 'string' ? { _type:'number[]', _array:'number' } : argTypes[0];
+	}
+
+	if (fnname == 'mkmodule') {
+		var m = modules[call.arguments[0].value];
+		if (!m) {
+			m = modules[call.arguments[0].value] = { _type:'table', _module:ctx };
+			if (call.arguments[0].value == 'plugins.eventful') {
```
Comment (pronvit) [inline index.js]: Any way not to hardcode all this here?

File: index.js [hunk 2]
```diff
@@ -698,49 +1348,102 @@ function find_comment_dfver(b)
...
+			var ctypes = c.types;
+
+			if (!b.isLocal && srcstack.length == 1) {
+				unchecked_global_fns[n] = b.loc.start.line;
+			}
+
+			var found = false;
+			if (base && base._defclass && member) {
+				for (var p = base; p; p = p._super) {
+					if (p._type == 'Screen' && (member == 'onShow' || member == 'onDismiss' || member == 'onDestroy' || member == 'onRender' || member == 'onIdle' || member == 'onHelp' || member == 'onGetSelectedUnit' || member == 'onGetSelectedItem' || member == 'onGetSelectedJob' || member == 'onGetSelectedBuilding')) {
```
Comment (pronvit) [inline index.js]: And this ideally shouldn't be hardcoded right in process() either.

File: index.js (hunk 1 above)
Comment (BenLubar) [reply index.js]: gonna move this to `builtins_basic.js`

File: index.js (hunk 2 above)
Comment (BenLubar) [reply index.js]: I have an `anyfunc:none` type that would work for this one, and the others can do similar things. I can also remove the `__gui_Painter` type alias.

**Pull request merged**